HLTriggerOffline/HeavyFlavor/src/HeavyFlavorValidation.cc

```diff
@@ -649,6 +648,71 @@ void HeavyFlavorValidation::myBook1D(DQMStore::IBooker & ibooker, TString name,
   delete h;
 }
 
+int HeavyFlavorValidation::getFilterLevel(const std::string &moduleName, const HLTConfigProvider &hltConfig)
+{
+  // helper lambda to check if a string contains a substring
+  const auto contains = [](const std::string &s, const std::string &sub) -> bool {
+    return s.find(sub) != std::string::npos;
+  };
+
+  // helper lambda to check if a string contains any of a list of substrings
+  const auto containsAny = [](const std::string &s, const std::vector<std::string> &subs) -> bool {
+    for (const auto &sub : subs) {
+      if (s.find(sub) != std::string::npos) return true;
+    }
+    return false;
+  };
+
+  // helper lambda to check if string s is any of the strings in vector ms
+  const auto isAnyOf = [](const std::string &s, const std::vector<std::string>& ms) -> bool {
+    for (const auto &m : ms) {
+      if (s == m) return true;
+    }
+    return false;
+  };
+
+  // tmadlener, 20.08.2017:
+  // define the valid module names for the different "levels", to add a little bit more stability
+  // to the checking compared to just doing some name matching.
+  // Note, that the name matching is not completely remved, since at level 4 and 5 some of the
+  // valid modules are the same, so that the name matching is still needed.
+  // With the current definition this yields the exact same levels as before, but weeds out some
+  // of the "false" positives at level 3 (naming matches also to some HLTMuonL1TFilter modules due to
+  // the 'forIterL3' in the name)
+  const std::string l1Filter = "HLTMuonL1TFilter";
+  const std::string l2Filter = "HLTMuonL2FromL1TPreFilter";
+  const std::vector<std::string> l3Filters = {"HLTMuonDimuonL3Filter", "HLTMuonL3PreFilter"};
+  const std::vector<std::string> l4Filters = {"HLTDisplacedmumuFilter", "HLTDiMuonGlbTrkFilter",
+                              "HLTMuonTrackMassFilter"};
+  const std::vector<std::string> l5Filters = {"HLTmumutkFilter", "HLT2MuonMuonDZ", "HLTDisplacedmumuFilter"};
+
+  if (contains(moduleName, "Filter") && hltConfig.moduleEDMType(moduleName) == "EDFilter") {
+    int level = -1; // if we get here, one of the following clauses should always catch!
+    if (contains(moduleName, "L1") && !contains(moduleName, "ForIterL3") &&
+        hltConfig.moduleType(moduleName) == l1Filter) {
+      level = 1;
+    }
+    if (contains(moduleName, "L2") && hltConfig.moduleType(moduleName) == l2Filter) {
```

Comment:
can this and subsequent if statements be else if instead? eg, once you have found a level value, the subsequent ifs do not need to be checked, right?